### PR TITLE
Update SmallRye Config to 3.2.0

### LINKED ISSUE #31402: Error opening config URL is ignored
### Describe the bug

When using `quarkus.config.locations` quarkus swallows errors when the listed URLs cannot be opened.

### Expected behavior

Quarkus should use the Fail Early paradigm and fail to start, throwing a clear error that states which given configuration location could not be found.
Optionally another config option could be added to enable ignoring failed config lookups, but this should not be the default.

### Actual behavior

Quarkus starts as if all is well and silently ignores the error that happend when trying to read the configuration.

### How to Reproduce?

Run any quarkus application like this:

```
java -Dquarkus.config.locations=https://example.com/clearly-wrong.properties -jar target\greeting-service-1.0.0-SNAPSHOT-runner.jar
```

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.13.5.Final-redhat-00002

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### LINKED ISSUE #31621: Read encrypted configuration from configuration sources
- From https://github.com/smallrye/smallrye-config/pull/833

### LINKED ISSUE #31759: config validation fails for Optional<List<CustomInterface>>
### Describe the bug

I'm trying to use the following configuration interface in my project:

```
@StaticInitSafe
@ConfigMapping(prefix = "some.key")
interface MyConfig {

    fun thing(): Optional<List<MySubConfig>>

}
interface MySubConfig {
    fun path(): String
    fun permission(): String
}

```

If bean validation is enabled by including `io.quarkus:quarkus-hibernate-validator` then config validation fails resulting in a startup failure. If I do not include the dependency then validation succeeds.

### Expected behavior

Config validation suceeds

### Actual behavior

If bean validation is enabled by including `io.quarkus:quarkus-hibernate-validator` then config validation fails with the following error message:
```
java.util.Optional incompatible with java.util.Collection
java.lang.ClassCastException: java.util.Optional incompatible with java.util.Collection
io.smallrye.config.validator.BeanValidationConfigValidator.validateProperty(BeanValidationConfigValidator.java:111)
```

### How to Reproduce?

Copy the config from above and include the `io.quarkus:quarkus-hibernate-validator` dependency.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-11.0.11+9 (build 11.0.11+9)
Eclipse OpenJ9 VM AdoptOpenJDK-11.0.11+9 (build openj9-0.26.0, JRE 11 Linux amd64-64-Bit Compressed References 20210421_975 (JIT enabled, AOT enabled)
OpenJ9   - b4cc246d9
OMR      - 162e6f729
JCL      - 7796c80419 based on jdk-11.0.11+9)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.13.5.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by radcortez**
- Fixes #31759
- Fixes #31402 
- Fixes #31621

### CODE REVIEW

**Comment by michalvavrik** [comment]
@radcortez regarding secret key handlers - users can use SmallRye Config documentation and you don't plan additional PR for docs etc., correct? (no objections, just checking as I need to know)

**Review by gsmet — APPROVED** [review]

File: bom/application/pom.xml [hunk 1]
```diff
@@ -54,8 +54,8 @@
         <microprofile-jwt.version>2.1</microprofile-jwt.version>
         <microprofile-lra.version>2.0</microprofile-lra.version>
         <microprofile-openapi.version>3.1</microprofile-openapi.version>
-        <smallrye-common.version>2.1.0</smallrye-common.version>
-        <smallrye-config.version>3.1.3</smallrye-config.version>
+        <smallrye-common.version>2.0.0</smallrye-common.version>
```
Comment (gsmet) [inline bom/application/pom.xml:57]: Is it normal you downgrade smallrye-common?

**Review by gsmet — CHANGES_REQUESTED** [review]

File: bom/application/pom.xml (hunk 1 above)
Comment (radcortez) [reply bom/application/pom.xml:57]: No. Probably a messed rebase.

**Comment by radcortez** [comment]
> @radcortez regarding secret key handlers - users can use SmallRye Config documentation and you don't plan additional PR for docs etc., correct? (no objections, just checking as I need to know)

I'll add docs to Quarkus too because users are not totally aware of SmallRye Config.

**Review by gsmet — APPROVED** [review]

**Pull request merged**